charms/kfp-api/src/charm.py

```diff
@@ -554,14 +585,20 @@ def _on_upgrade(self, _):
     def _on_remove(self, _):
         """Remove all resources."""
         self.unit.status = MaintenanceStatus("Removing K8S resources")
-        k8s_resources_manifests = self.k8s_resource_handler.render_manifests()
-        try:
-            delete_many(self.k8s_resource_handler.lightkube_client, k8s_resources_manifests)
-        except ApiError as error:
-            # do not log/report when resources were not found
-            if error.status.code != 404:
-                self.logger.error(f"Failed to delete K8S resources, with error: {error}")
-                raise error
+        # Define the KubernetesResourceHandler manually because the object storage conflict in
+        # `self._context` is not available anymore, and all we need is the labels/resource_types
+        # to delete the resources anyway
+        k8s_resource_handler = KubernetesResourceHandler(
+            field_manager=self._lightkube_field_manager,
+            template_files=[],
+            context={},
+            logger=self.logger,
+            labels=self._charm_default_kubernetes_labels,
+            resource_types={Service, ServiceAccount, ClusterRole, ClusterRoleBinding},
+        )
+        load_in_cluster_generic_resources(k8s_resource_handler.lightkube_client)
+        k8s_resource_handler.delete()
```

Comment:
This change is covered in #385, we should track this change there

Reply:
This change is tracked in #385.  This PR is based on #385.  When #385 is merged, this should no longer show up as a change.

Reply:
I think it is clearer now that things have been rebased.